l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -656,13 +656,30 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
     /// @dev IL1AssetHandler interface, used to undo a failed migration of a chain.
     /// @param _chainId the chainId of the chain
     /// @param _assetId the assetId of the chain's STM
-    /// @param _data the data for the recovery
+    /// @param _data the data for the recovery. The structure for this data is:
+    ///              depositAmount || l2Receiver || chainId || stmData || chainData
```

Comment:
Could you decode the data in this function, and only pass the stmData to the STM and the chainData to the chain? like it is done in bridgeMint

Reply:
@koloz193 could you please provide the link somewhere? From what I see this is `bytes memory _assetData` aka `transferData` from the function `bridgehubDeposit`  and for chain migration it does not need deposit amount and l2 receiver

Reply:
https://github.com/matter-labs/era-contracts/blob/3620cfe7d7e8b5446ac5a9afdf266449c48b55c2/l1-contracts/contracts/bridge/L1AssetRouter.sol#L753 which is called from https://github.com/matter-labs/era-contracts/blob/3620cfe7d7e8b5446ac5a9afdf266449c48b55c2/l1-contracts/contracts/bridge/L1AssetRouter.sol#L513

Reply:
it is `LEGACY_ENCODING_VERSION`

Reply:
for chain migration the `NEW_ENCODING_VERSION` is used